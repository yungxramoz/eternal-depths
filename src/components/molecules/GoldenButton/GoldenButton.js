import React from 'react'
import './GoldenButton.css'
import RpgContainer from '../../templates/RpgContainer/RpgContainer'

const GoldenButton = ({ children, onClick, disabled, className = '' }) => {
  return (
    <button
      className={`golden-button ${className}`}
      onClick={onClick}
      disabled={disabled}
    >
      <RpgContainer golden2 className="golden-container">
        {children}
      </RpgContainer>
    </button>
  )
}

export default GoldenButton
