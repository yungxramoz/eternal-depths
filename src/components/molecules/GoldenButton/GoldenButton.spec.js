import { render, screen } from '@testing-library/react'
import React from 'react'
import GoldenButton from './GoldenButton'

describe('GoldenButton', () => {
  it('renders correctly', () => {
    render(
      <GoldenButton className="mock-class">
        <span>Test</span>
      </GoldenButton>,
    )
    expect(document.body).toMatchSnapshot('GoldenButton')
  })
  it('renders correctly when disabled', () => {
    render(
      <GoldenButton disabled>
        <span>Test</span>
      </GoldenButton>,
    )
    expect(document.body).toMatchSnapshot('GoldenButton disabled')
  })
  it('triggers onClick', () => {
    const onClick = jest.fn()
    render(
      <GoldenButton onClick={onClick}>
        <span>Test</span>
      </GoldenButton>,
    )
    screen.getByText('Test').click()
    expect(onClick).toHaveBeenCalled()
  })
})
