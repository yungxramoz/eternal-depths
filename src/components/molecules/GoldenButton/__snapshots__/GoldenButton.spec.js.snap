// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoldenButton renders correctly when disabled: GoldenButton disabled 1`] = `
<body>
  <div>
    <button
      class="golden-button "
      disabled=""
    >
      <div
        class="rpgui-container container framed framed-golden-2 golden-container"
      >
        <span>
          Test
        </span>
      </div>
    </button>
  </div>
</body>
`;

exports[`GoldenButton renders correctly: GoldenButton 1`] = `
<body>
  <div>
    <button
      class="golden-button mock-class"
    >
      <div
        class="rpgui-container container framed framed-golden-2 golden-container"
      >
        <span>
          Test
        </span>
      </div>
    </button>
  </div>
</body>
`;
